app/components/autocomplete/emoji_suggestion/emoji_suggestion.js

```diff
@@ -63,18 +72,19 @@ export default class EmojiSuggestion extends Component {
             return;
         }
 
-        const matchTerm = match[3];
+        const oldMatchTerm = this.matchTerm;
+        this.matchTerm = match[3];
 
-        const matchTermChanged = matchTerm !== this.state.matchTerm;
-        if (matchTermChanged) {
-            this.setState({
-                matchTerm
-            });
+        const pre47 = !isMinimumServerVersion(Client4.getServerVersion(), 4, 7);
+
+        if (!pre47 && this.matchTerm !== oldMatchTerm && this.matchTerm.length) {
```

Comment:
Is it possible to break out this logic so we have something like
```
if (pre47) {
    // do one thing
} else {
    // do a similar, but slightly different thing
}
```
? It'll probably result in a bit of code duplication, but it'll make this easier to follow since I'm having a tough time figuring this out